verl/models/mcore/model_initializer.py

```diff
@@ -50,31 +56,94 @@ def init_mcore_model_dense(
 
 
 def init_mcore_model_qwen2_moe(
-    tfconfig, hf_config, pre_process=None, post_process=None, share_embeddings_and_output_weights=False, value=False
+    tfconfig,
+    hf_config,
+    pre_process=None,
+    post_process=None,
+    share_embeddings_and_output_weights=False,
+    value=False,
+    freeze_moe_router=True,
+    **extra_kwargs,
 ):
-    return init_mcore_model_dense(
-        tfconfig, hf_config, pre_process, post_process, share_embeddings_and_output_weights, value
+    from megatron.core.models.gpt.gpt_layer_specs import get_gpt_decoder_block_spec
+    from megatron.core.models.gpt.gpt_model import GPTModel
+
+    use_te = True
+    if freeze_moe_router:
+        tfconfig.moe_router_load_balancing_type = "none"
+
+    def patch_layer_spec(transformer_layer_spec):
+        # shared_experts.gate=True
+        for i in range(len(transformer_layer_spec.layer_specs)):
+            transformer_layer_spec.layer_specs[i].submodules.mlp.submodules.shared_experts.params["gate"] = True
+        return transformer_layer_spec
+
+    assert tfconfig.normalization == "RMSNorm", "only RMSNorm is supported for now"
+    transformer_layer_spec = get_gpt_decoder_block_spec(tfconfig, use_transformer_engine=use_te)
```

Comment:
nit: can we directly set `use_transformer_engine` to True instead of using one variable `use_te`?